homeassistant/components/renson/binary_sensor.py

```diff
@@ -0,0 +1,141 @@
+
+
+BINARY_SENSORS: tuple[RensonBinarySensorEntityDescription, ...] = (
+    RensonBinarySensorEntityDescription(
+        name="Frost protection active",
+        key="FROST_PROTECTION_FIELD",
+        field=FROST_PROTECTION_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="BREEZE_ENABLE_FIELD",
+        name="Breeze",
+        field=BREEZE_ENABLE_FIELD,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="BREEZE_MET_FIELD",
+        name="Breeze conditions met",
+        field=BREEZE_MET_FIELD,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="HUMIDITY_CONTROL_FIELD",
+        name="Humidity control",
+        field=HUMIDITY_CONTROL_FIELD,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="AIR_QUALITY_CONTROL_FIELD",
+        name="Air quality control",
+        field=AIR_QUALITY_CONTROL_FIELD,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="CO2_CONTROL_FIELD",
+        name="CO2 control",
+        field=CO2_CONTROL_FIELD,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="PREHEATER_FIELD",
+        name="Preheater",
+        field=PREHEATER_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+
+class RensonBinarySensor(RensonEntity, BinarySensorEntity):
+    """Get a sensor data from the Renson API and store it in the state of the class."""
+
+    def __init__(
+        self,
+        description: RensonBinarySensorEntityDescription,
+        api: RensonVentilation,
+        coordinator: RensonCoordinator,
+    ) -> None:
+        """Initialize class."""
+        super().__init__(description.key, api, coordinator)
+
+        self.field = description.field
+        self.entity_description = description
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        all_data = self.coordinator.data
+
+        value = self.api.get_field_value(all_data, self.field.name)
+
+        self._attr_is_on = self.api.parse_value(value, DataType.BOOLEAN)
+
+        self.async_write_ha_state()
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Call the Renson integration to setup."""
+
+    api: RensonVentilation = hass.data[DOMAIN][config_entry.entry_id]["api"]
+    coordinator: RensonCoordinator = hass.data[DOMAIN][config_entry.entry_id][
+        "coordinator"
+    ]
+
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
first refresh should happen inside `renson/__init__.py` because each platform relies on it.